src/main/java/org/janelia/saalfeldlab/n5/StringDataBlock.java

```diff
@@ -55,9 +55,13 @@ public ByteBuffer toByteBuffer() {
 
     @Override
     public void readData(final ByteBuffer buffer) {
-        if (buffer.array() != serializedData)
-            buffer.get(serializedData);
-        actualData = deserialize(buffer.array());
+
+		if (buffer.hasArray()) {
+			if (buffer.array() != serializedData)
+				buffer.get(serializedData);
+			actualData = deserialize(buffer.array());
+		} else
+			actualData = ENCODING.decode(buffer).toString().split(NULLCHAR);
```

Comment:
```suggestion
                byte[] buffer_array;
		if (buffer.hasArray()) {
		        // obtain existing array that backs the buffer
		        buffer_array = buffer.array()
		} else {
		        // allocate a new array and store the data into the buffer
		        buffer_array = new byte[buffer.position()];
		        buffer.get(buffer_array);
		}
	        if (buffer_array != serializedData)
			actualData = deserialize(buffer_array);
```

Reply:
Do you have a reason for preferring that implementation? 

I was preferring not to explicitly allocate a new array. Ideally we should benchmark these two, but I don't expect there will be much of a performance difference.

@axtimwalde do you have a preference?

Reply:
I agree with @mkitti here that the deserialization logic should be separate from `readData`. But In my view, just providing and overload `protected String[] deserialize(ByteBuffer buffer)` would be fine.

Reply:
Had a look with @axtimwalde just now, and talked briefly irl w/ @minnerbe 

> the deserialization logic should be separate from readData

Agreed, this would be nice, but is one of several considerations.  The logic is `.split(NULLCHAR)`, which is not so much, and I will be happy to have duplicated logic for better performance (fewer arrays allocated).

The difference between what this PR currently has and @mkitti 's edit (as I see it) is
```
ENCODING.decode(buffer)
```
vs 
```
// allocate a new array and store the data into the buffer
buffer_array = new byte[buffer.position()];
buffer.get(buffer_array);
```
The latter definitely allocates a new `byte[]`, the former may or may not depending on how `decode` is implemented.  @axtimwalde and I prefer "possibly-re-use-data" to "definitely don't re-use" data.

We also considered replacing the whole readData method with 
```
actualData = ENCODING.decode(buffer).toString().split(NULLCHAR);
```
which also would address the issue. But in this case we prefer the "definitely re-use data" of @minnerbe 's initial implementation to the "possibly re-use data of `ENCODING.decode`. 

The conclusion being that we'll stick with whats in the PR.